### PR TITLE
move to flat config

### PR DESCRIPTION
**Pull request by noamokman**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,34 +44,33 @@
...
     "npm-run-all": "^4.1.5",
-    "prettier": "^3.3.2",
-    "prettier-plugin-sort-json": "^4.0.0",
-    "typescript": "^5.4.5"
+    "prettier": "^3.3.3",
+    "typescript": "^5.5.3"
   },
   "peerDependencies": {
-    "eslint": "^8.57.0"
+    "eslint": "^9.0.0"
```
[the remark this answers is not in the document]
Comment (noamokman) [reply package.json:62]: https://eslint.org/docs/latest/extend/shareable-configs#publishing-a-shareable-config

File: eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+    },
+    parser: typescriptEslint.parser,
+  },
+  plugins: {
+    unicorn: unicornPlugin,
+    'import-x': importXPlugin,
+    '@typescript-eslint': typescriptEslint.plugin,
+  },
+  rules: {
+    '@typescript-eslint/no-unnecessary-type-parameters': 'error',
```
[the remark this answers is not in the document]
Comment (noamokman) [reply eslint.config.js:25]: I agree though about it too, not sure how we can easily do that

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,572 @@
+// @ts-check
```
[the remark this answers is not in the document]
Comment (noamokman) [reply eslint.config.js:1]: this is the standard for eslint config files but it doesn't really matter

**Comment by noamokman** [comment]
Thanks for the review,
I did also deprecate `react-native` and I did change the test rule to use `vitest` instead of `jest`

**Review by SharonGrossman — APPROVED** [review]

**Pull request merged**